### PR TITLE
add security advisory for the authz policy

### PR DESCRIPTION
**Pull request by yangminzhu**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Review by jacob-delgado — APPROVED** [review]

File: content/en/news/security/istio-security-2021-003/index.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    - source:
+        notNamespaces: ["*"]
+{{< /text >}}
+
+Also check the [security policy examples](/docs/ops/configuration/security/security-policy-examples/#require-mtls-in-authorization-layer-defense-in-depth)
+for more details about the above alternative solution.
+
+## Credit
+
+We'd like to thank [John Howard](https://github.com/howardjohn/) for reporting this issue.
```
Comment (howardjohn) [inline content/en/news/security/istio-security-2021-003/index.md:132]: Thanks :slightly_smiling_face:

**Review by howardjohn — APPROVED** [review]
LGTM

**Review by nrjpoddar — APPROVED** [review]
Thanks @yangminzhu for iterating on this. Looks really good!

File: content/en/news/security/istio-security-2021-003/index.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+- Your authorization policy uses mTLS-only fields and you have also enabled mTLS with STRICT mode or your authorization
+policy is configured to reject plain text traffic explicitly.
+
+## Issue
+
+In authorization policy, the following are mTLS-only fields:
+
+- the principals field under the source section
+- the namespaces field under the source section
```
Comment (ericvn) [inline content/en/news/security/istio-security-2021-003/index.md:31]: The examples also show `notNamespaces`. Should this be called out in addition to the `namespaces` field?

File: content/en/news/security/istio-security-2021-003/index.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+You can stop reading if
+
+- Your authorization policy does not use [mTLS-only fields](/docs/concepts/security/#dependency-on-mutual-tls); or
+
+- Your authorization policy uses mTLS-only fields and you have also enabled mTLS with STRICT mode or your authorization
+policy is configured to reject plain text traffic explicitly.
+
+## Issue
+
+In authorization policy, the following are mTLS-only fields:
```
Comment (ericvn) [inline content/en/news/security/istio-security-2021-003/index.md:28]: In the first paragraph we have a links for mTLS-only fields, which also has an additional field. Maybe that's fine, but wanted to call it out here as a difference. Would it make sense to create a link within this document to the fields (add link here), and use that above?

Comment (yangminzhu) [reply content/en/news/security/istio-security-2021-003/index.md]: Good point, actually I think the additional field `connection.sni` should be removed from the page as it does not depend on mutual TLS (it only requires TLS).

File: content/en/news/security/istio-security-2021-003/index.md (hunk 2 above)
Comment (yangminzhu) [reply content/en/news/security/istio-security-2021-003/index.md]: good suggestion, I actually updated the example to make it shorter and also extended the explanation to make it more clear.

File: content/en/news/security/istio-security-2021-003/index.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+keywords: [CVE]
+skip_seealso: true
+---
+
+{{< security_bulletin >}}
+
+This is a security advisory for customers to check the authorization policy to make sure [mTLS (STRICT mode) is enabled](/docs/tasks/security/authentication/authn-policy/#globally-enabling-istio-mutual-tls-in-strict-mode)
+when using [mTLS-only fields](/docs/concepts/security/#dependency-on-mutual-tls) in the authorization policy.
+
+You can stop reading if
```
Comment (ericvn) [inline content/en/news/security/istio-security-2021-003/index.md:19]: Nit: Add : at end of line.

File: content/en/news/security/istio-security-2021-003/index.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+spec:
+  action: ALLOW
+  rules:
+  - from:
+    - source:
+        notNamespaces: ["foo"]
+{{< /text >}}
+
+A **plain text request** from the namespace `foo` will actually be allowed. The mTLS-only field `notNamespaces` will be
+compared to an empty value when mTLS is not used. The comparison result is always matched that makes the policy to allow
```
Comment (ericvn) [inline content/en/news/security/istio-security-2021-003/index.md:53]: The last sentence (similar in section below) is slightly confusing.  Maybe something like:
```
A **plain text request** from the namespace `foo` will actually be allowed. The mTLS-only field
`notNamespaces` will be  compared to an empty value when mTLS is not used, resulting in a
policy that allows all requests that are not from an empty namespace. 
```

File: content/en/news/security/istio-security-2021-003/index.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+spec:
+  action: DENY
+  rules:
+  - from:
+    - source:
+        namespaces: ["foo"]
+{{< /text >}}
+
+A **plain text request** from the namespace `foo` will not be rejected. The mTLS-only field `namespaces` will be
+compared to an empty value when mTLS is not used. The comparison result is always not matched that makes the policy to
```
Comment (ericvn) [inline content/en/news/security/istio-security-2021-003/index.md:72]: Again, the last sentence is a little confusing.

File: content/en/news/security/istio-security-2021-003/index.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+not reject the request as if it was not from the namespace `foo`.
+
+## Solution
+
+To solve this problem, it's recommended to always [enable mTLS with STRICT mode](/docs/tasks/security/authentication/authn-policy/#enable-mutual-tls-per-namespace-or-workload)
+on the workloads before using any mTLS-only fields in the authorization policy on the same workload.
+
+If you are unable to enable mTLS with STRICT mode for the workload, the alternative solution is to update the authorization
+policy to explicitly allow traffic with non-empty namespaces or reject traffic with empty namespaces (`*` implies non-empty and `not *` implies empty).
+As namespace can only be extracted when mTLS is STRICT. The policies below effectively also reject any plain text traffic.
```
Comment (ericvn) [inline content/en/news/security/istio-security-2021-003/index.md:82]: I think you can either remove the `As namespace...` sentence (it's not a full sentence), or end the prior sentence with a comma and lowercase As. That does make the prior sentence a little too long.  Maybe:

```
If you are unable to enable mTLS with STRICT mode for the workload, the alternative solution is
to update the authorization policy to explicitly allow traffic with non-empty namespaces or
reject traffic with empty namespaces, as namespace can only be extracted when mTLS is 
STRICT. `*` implies non-empty namespaces and `not *` implies empty namespaces. The 
policies below also reject any plain text traffic.
```

File: content/en/news/security/istio-security-2021-003/index.md (hunk 5 above)
Comment (yangminzhu) [reply content/en/news/security/istio-security-2021-003/index.md]: updated, I changed the last part a little bit to make it more accurate that it only applies to plain text traffic.

File: content/en/news/security/istio-security-2021-003/index.md (hunk 6 above)
Comment (yangminzhu) [reply content/en/news/security/istio-security-2021-003/index.md]: updated.

**Review by ericvn — APPROVED** [review]

**Pull request merged**